ddl/ddl_api.go

```diff
@@ -3251,6 +3252,30 @@ func (d *ddl) AddTablePartitions(ctx sessionctx.Context, ident ast.Ident, spec *
 		return errors.Trace(err)
 	}
 
+	if d.IsTiFlashPollEnabled() && meta.TiFlashReplica != nil {
+		// Must set placement rule before the ActionAddTablePartition job is in queue.
+		tikvStore, ok := ctx.GetStore().(helper.Storage)
+		if !ok {
+			logutil.BgLogger().Error("can not get Helper")
+		} else {
+			tikvHelper := &helper.Helper{
+				Store:       tikvStore,
+				RegionCache: tikvStore.GetRegionCache(),
+			}
```

Comment:
I suggest you adding another `tiflashManager` in `infosync` package, and move methods like `SetPlacementRule` to that interface. You don't need to know anything about tikv to do placement rules.